components/datetime/src/format/neo.rs

```diff
@@ -0,0 +1,1325 @@
+    /// );
+    /// ```
+    pub fn format_time<'l, T>(
+        &'l self,
+        pattern: &'l Pattern<'l>,
+        datetime: &T,
+    ) -> FormattedDateTimePattern<'l>
+    where
+        T: IsoTimeInput,
+    {
+        FormattedDateTimePattern {
+            pattern,
+            datetime: ExtractedDateTimeInput::extract_from_time(datetime),
+            interpolator: self.as_borrowed(),
+        }
+    }
+}
+
+/// A pattern that has been interpolated and implements [`Writeable`].
+///
+/// <div class="stab unstable">
+/// 🚧 This code is experimental; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. It can be enabled with the "experimental" Cargo feature
+/// of the icu meta-crate. Use with caution.
+/// <a href="https://github.com/unicode-org/icu4x/issues/1317">#1317</a>
+/// </div>
+#[derive(Debug)]
+pub struct FormattedDateTimePattern<'l> {
+    pattern: &'l Pattern<'l>,
+    datetime: ExtractedDateTimeInput,
+    interpolator: RawDateTimePatternInterpolatorBorrowed<'l>,
+}
+
+impl<'l> Writeable for FormattedDateTimePattern<'l> {
+    fn write_to<W: fmt::Write + ?Sized>(&self, sink: &mut W) -> fmt::Result {
+        let loc_datetime =
+            DateTimeInputWithWeekConfig::new(&self.datetime, self.interpolator.week_calculator);
+        write_pattern(
+            self.pattern,
+            Some(&self.interpolator),
+            Some(&self.interpolator),
+            &loc_datetime,
+            self.interpolator.fixed_decimal_formatter,
+            sink,
+        )
+        .map_err(|_e| {
+            icu_provider::_internal::log::warn!("{_e:?}");
+            core::fmt::Error
+        })
+    }
+
+    // TODO(#489): Implement writeable_length_hint
+}
+
+impl<'l> fmt::Display for FormattedDateTimePattern<'l> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.write_to(f)
+    }
+}
+
+impl<'data> DateSymbols<'data> for RawDateTimePatternInterpolatorBorrowed<'data> {
+    fn get_symbol_for_month(
+        &self,
+        field_symbol: fields::Month,
+        field_length: fields::FieldLength,
+        code: MonthCode,
+    ) -> Result<MonthPlaceholderValue, Error> {
+        let field = fields::Field {
+            symbol: fields::FieldSymbol::Month(field_symbol),
+            length: field_length,
+        };
+        let month_symbols = self
+            .month_symbols
+            .get_with_length(field_symbol, field_length)
+            .ok_or_else(|| Error::MissingNames(field))?;
+        let Some((month_number, is_leap)) = code.parsed() else {
+            return Err(Error::MissingMonthSymbol(code));
+        };
+        let Some(month_index) = month_number.checked_sub(1) else {
+            return Err(Error::MissingMonthSymbol(code));
+        };
+        let month_index = usize::from(month_index);
+        let symbol = match month_symbols {
+            MonthSymbolsV1::Linear(linear) => {
+                if is_leap {
+                    None
```

Comment:
nit: maybe emit a warning? hm, unsure.

Reply:
Note: It becomes a `Error::MissingMonthSymbol(code)`, which later gets logged as a warning (#4336)

Reply:
sgtm